External/FEXCore/Source/Interface/Core/OpcodeDispatcher.h

```diff
@@ -1040,19 +1060,105 @@ friend class FEXCore::IR::PassManager;
   [[nodiscard]] uint32_t GetDstBitSize(X86Tables::DecodedOp Op) const;
   [[nodiscard]] uint32_t GetSrcBitSize(X86Tables::DecodedOp Op) const;
 
+  static inline unsigned IndexNZCV(unsigned BitOffset) {
+    switch (BitOffset) {
+      case FEXCore::X86State::RFLAG_OF_LOC: return 28;
+      case FEXCore::X86State::RFLAG_CF_LOC: return 29;
+      case FEXCore::X86State::RFLAG_ZF_LOC: return 30;
+      case FEXCore::X86State::RFLAG_SF_LOC: return 31;
+      default: assert(0);
+    }
+  }
+
+  OrderedNode *GetNZCV() {
+    if (!CachedNZCV) {
+      CachedNZCV = _LoadFlag(FEXCore::X86State::RFLAG_NZCV_LOC);
+
+      // We don't know what's set
+      PossiblySetNZCVBits = ~0;
+    }
+
+    return CachedNZCV;
+  }
+
+  void SetNZCV(OrderedNode *Value) {
+    CachedNZCV = Value;
+  }
+
+  void ZeroNZCV() {
+    CachedNZCV = _Constant(0);
+    PossiblySetNZCVBits = 0;
+  }
+
+  void SetN_ZeroZCV(unsigned SrcSize, OrderedNode *Res) {
+    assert(IndexNZCV(FEXCore::X86State::RFLAG_SF_LOC) == 31);
```

Comment:
```suggestion
    LOGMAN_THROW_A_FMT(IndexNZCV(FEXCore::X86State::RFLAG_SF_LOC) == 31, "Insert funny message here");
```

Just so it's using our assertion interface

Reply:
What's.. what's wrong with assert?

Reply:
The logman assert interface will actually log the message as to why it failed to FEXServer so we can see why it failed outside of having running through gdb. Which is a nicety.
Also assert doesn't actually fail, because the guest application might capture the signal and try to keep running. Logman will execute the `AssertAndDie` routine that makes it know that FEX /really/ /really/ wants to die.

Reply:
Although in this case, you could likely change the function signature to be `static inline constexpr unsigned IndexNZCV(unsigned BitOffset)` and then after the function declaration just use `static_asssert` with each of the types, that way it gets enforced at compile time.

Reply:
I like that better ^^